### ISSUE TITLE
Training data not shuffled and truncated in Triplet Model training pipeline

### ISSUE THREAD

**Issue by VecherVhatuX** [VecherVhatuX/learning_data_science_in_prison#594]
Triplet Model Training Pipeline Performance Issue

The current implementation of the Triplet Model training pipeline has two major issues. The training data is not shuffled at the end of each epoch, leading to overfitting and poor model generalization. Additionally, the training dataset is truncated to a multiple of the batch size, resulting in unused training data.
